src/algebra/free_monoid.lean

```diff
@@ -119,4 +120,15 @@ hom_eq $ λ x, rfl
 lemma map_comp (g : β → γ) (f : α → β) : map (g ∘ f) = (map g).comp (map f) :=
 hom_eq $ λ x, rfl
 
+instance : star_monoid (free_monoid α) :=
+{ star := list.reverse,
+  star_involutive := list.reverse_reverse,
+  star_mul := list.reverse_append, }
+
+@[simp]
+lemma star_of (x : α) : star (of x) = of x := rfl
+
+@[simp]
+lemma star_one : star (1 : free_monoid α) = 1 := rfl
```

Comment:
I'm confused... shouldn't this be an axiom for `star_monoid`?

Reply:
It's implied by the multiplication and involutivity axioms, although I don't know if it's a simp lemma.

Reply:
I added it here in case there was value to "this is not only true, but definitionally true". Perhaps there is no value to that.

Reply:
Should I remove this, since the global `star_one` already covers it?

Reply:
I guess, because it is defeq, there is some value in it. Because now `dsimp` can use it.
Maybe add a comment, to explain that is why you want to add it?